### DISCUSSION TITLE
[错误报告]: 1.1.2版本动漫独立两级目录复发

### DISCUSSION THREAD

**Discussion by Smart-zsw** [jxxghp/MoviePilot#422] [讨论]
### 确认

- [X] 我的版本是最新版本，我的版本号与 [version](https://github.com/jxxghp/MoviePilot/releases/latest) 相同。
- [X] 我已经 [issue](https://github.com/jxxghp/MoviePilot/issues) 中搜索过，确认我的问题没有被提出过。
- [X] 我已经 [Telegram频道](https://t.me/moviepilot_channel) 中搜索过，确认我的问题没有被提出过。
- [X] 我已经修改标题，将标题中的 描述 替换为我遇到的问题。

### 当前程序版本

1.1.2

### 问题类型

主程序运行问题

### 问题描述

问题如题

**Comment by dlovew** [comment]
你不想动漫单独分类，在分类配置文件改一下就是！

**Comment by dlovew** [comment]
像我们需要动漫二级分类的怎么搞？

**Comment by Smart-zsw** [comment]
你不填写媒体库动漫的环境信息，就自动到电视剧目录下的动漫目录了，我这个是动漫独立目录，生成了两级，重复了

**Comment by dlovew** [comment]
> 你不填写媒体库动漫的环境信息，就自动到电视剧目录下的动漫目录了，我这个是动漫独立目录，生成了两级，重复了

我也是动漫独立目录，没有问题，你贴一下你的分类文件看看啊，你图也没有

**Comment by Smart-zsw** [comment]
![image](https://github.com/jxxghp/MoviePilot/assets/98108711/c32fd127-a272-4e56-9380-8e107a19e419)
![image](https://github.com/jxxghp/MoviePilot/assets/98108711/4c988af6-06d0-44fa-9a82-ea6495bde4b7)

**Comment by Smart-zsw** [comment]
影视剧都正常

**Comment by jxxghp** [comment]
想要动漫一级目录，且又开了二级分类的，把电视剧下面的二级分类删掉。

**Reply by Smart-zsw** [reply]
是删了的
![image](https://github.com/jxxghp/MoviePilot/assets/98108711/8392a055-a750-40d7-a1f6-0319065da271)

**Reply by dlovew** [reply]
理论上动漫单独设置了一级分类，电视剧下就算设置了动漫分类也不生效吧

**Comment by dlovew** [comment]
你可以把媒体库动漫名字这个环境变量删掉试试

**Reply by Smart-zsw** [reply]
那就变成默认了，默认是在电视剧下

**Comment by dlovew** [comment]
你把你分类配置文件发出来看看！不是发这个环境变量！

**Comment by dlovew** [comment]
因为我没问题，我怀疑你还去分类配置文件有错误

**Reply by Smart-zsw** [reply]
![image](https://github.com/jxxghp/MoviePilot/assets/98108711/26da9d87-5613-41b8-abd4-32aa6e97f4ff)
我一直没有动过他

**Comment by mjsoftking** [comment]
@Smart-zsw 是目录监控转移还是下载监控转移，如果是目录监控转移把文件夹映射发出来看看

**Reply by Smart-zsw** [reply]
我就是手动整理的，默认转移到媒体库

**Reply by mjsoftking** [reply]
![图片](https://github.com/jxxghp/MoviePilot/assets/23021469/8494270d-77ce-4db2-bee4-304dead71650)
我这个也是手动整理的，我没开二级分类。


如果可以把你的配置 TV下的动漫注释掉，然后手动整理试一下看看，如果只有一层动漫了，那可能是代码判断有点问题

**Reply by Smart-zsw** [reply]
下面的动漫的genreid也需要注释掉嘛

**Reply by dlovew** [reply]
或者分类文件里单独给动漫设置一下！我就是这样设置的，没毛病

**Reply by mjsoftking** [reply]
是的，这两个字段是一组的，注释一个就会导致文件错误了，就最前面加个 # 就行

**Reply by Smart-zsw** [reply]
那两个字段我都注释，变成这样了，我试试加一个动漫吧
![image](https://github.com/jxxghp/MoviePilot/assets/98108711/a5039eeb-bb10-483c-bf9d-a593babaab7b)

**Reply by mjsoftking** [reply]
那应该就是本身检测到是动漫，走到了动漫的库，然后因为开启了二级分类，他就在配置文件中检测了一下，又找到了一个属于动漫的分类，所以出了2个动漫文件夹，而修改后没有找到属于动漫但是匹配到了国产剧分类规则，所以应该属于是代码上判断出现一点问题

**Reply by Smart-zsw** [reply]
都试，没有用，识别成国产剧，哎，我就老实用电视剧下面的动漫吧，放弃了，谢谢两位大佬解答

**Reply by mjsoftking** [reply]
敢不敢尝试一下TV下动漫那个文本删除（动漫这2个字），保留冒号，这么改不一定能启动，可以尝试一下

**Reply by mjsoftking** [reply]
用单集试一下，别整体处理，怕出问题

**Reply by Smart-zsw** [reply]
我试第二种就是把动漫那两个字段，添加到底下跟tv同级，剪切过去的，没有用，他还会在动漫下面建立国产剧分类

**Reply by mjsoftking** [reply]
等 dlovew 老板把配置贴出来用用把，你的那个估计是定义写错了

**Comment by dlovew** [comment]
我在外面，回头把我的配置贴出来

**Comment by dlovew** [comment]
下载目录
![screenshot20230903](https://github.com/jxxghp/MoviePilot/assets/59035223/279252da-b043-4817-a441-8f78aa5e6eeb)
媒体库目录
![screenshot20230903 (1)](https://github.com/jxxghp/MoviePilot/assets/59035223/8c80c208-836d-4a84-91bb-86716d46b4ae)
文件夹设置
![screenshot20230903 (2)](https://github.com/jxxghp/MoviePilot/assets/59035223/0837c990-fc79-4f74-b083-f704625691a9)
![screenshot20230903 (3)](https://github.com/jxxghp/MoviePilot/assets/59035223/720b95aa-789c-4705-ae56-3f5ff04e8aab)


分类配置
movie:
  日本动画:
    original_language: 'ja'
    genre_ids: '16'
  国产动画:
    original_language: 'zh,cn,bo,za'
    genre_ids: '16'
  韩国电影:
    original_language: 'ko'
  欧美电影:
    original_language: 'en,fr,gd,de,es,it,nl,pt,ru,be,uk,th,cz,cs'
  日本电影:
    original_language: 'ja'
  华语电影:
    original_language: 'zh,cn,bo,za'
  儿童电影:
    genre_ids: '10762'
  未分类电影:

tv:
  日番:
    genre_ids: '16'
    original_language: 'ja'
  国漫:
    original_language: 'zh,cn,bo,za'
    genre_ids: '16'
  儿童节目:
    genre_ids: '10762'
  纪录片:
    genre_ids: '99'
  综艺节目:
    genre_ids: '10764,10767'
  大陆剧:
    origin_country: 'CN'
  港台剧:
    origin_country: 'HK,TW'
  日韩剧:
    origin_country: 'KP,KR,JP'
  欧美剧:
    origin_country: 'US,FR,GB,DE,IT,NL,PT,RU,UK,TH,ES'
  未分类电视剧:
comic:
  日番:
    genre_ids: '16'
    original_language: 'ja'
  国漫:
    original_language: 'zh,cn,bo,za'
    genre_ids: '16'

**Comment by Smart-zsw** [comment]
感谢大佬，我用不上分类这么细，我就设置个动漫就行，我就电视剧里用动漫就行，也能用，感谢帮助，看你这个配置我懂了，等我需要就知道怎么配置了
